File: app/src/main/java/com/coppermobile/mysamplemvvmdatabindinglivedata/data/source/local/AppDatabase.java
package com.coppermobile.mysamplemvvmdatabindinglivedata.data.source.local;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.Room;
import android.arch.persistence.room.RoomDatabase;
import android.content.Context;

import com.coppermobile.mysamplemvvmdatabindinglivedata.data.Comments;
import com.coppermobile.mysamplemvvmdatabindinglivedata.data.Dish;

@Database(entities = {Comments.class, Dish.class}, version = 1, exportSchema = false)
public abstract class AppDatabase extends RoomDatabase {

    private static AppDatabase instance;
    private static final String COMMENT_DATABASE_NAME = "comments_db";

    public abstract CommentsDAO commentsDoaMethod();
    public abstract DishDAO dishDaoMethod();

    public static AppDatabase getInstance(Context context) {
        if (instance == null) {
            try {
                instance = Room.databaseBuilder(context.getApplicationContext(), AppDatabase.class, COMMENT_DATABASE_NAME).build();
            } catch (Exception ignored) {
            }
        }
        return instance;
    }
}

File: app/src/main/java/com/coppermobile/kotlinsamplemvvmlivedataroom/data/source/local/AppDatabase.kt
package com.coppermobile.kotlinsamplemvvmlivedataroom.data.source.local

import android.arch.persistence.room.Database
import android.arch.persistence.room.Room
import android.arch.persistence.room.RoomDatabase
import android.content.Context
import android.util.Log
import com.coppermobile.kotlinsamplemvvmlivedataroom.data.Comments
import com.coppermobile.kotlinsamplemvvmlivedataroom.data.Dish

@Database(entities = [Dish::class, Comments::class], version = 1, exportSchema = false)
abstract class AppDatabase : RoomDatabase() {

    abstract fun getDishDao(): DishDao
    abstract fun getCommentsDao(): CommentsDao

    companion object {

        private var instance: AppDatabase? = null
        private const val DATABASE_NAME = "myDb"

        fun getInstance(context: Context): AppDatabase {
            if (instance == null) {
                synchronized(this) {}
                try {
                    instance = Room.databaseBuilder(context.applicationContext, AppDatabase::class.java, DATABASE_NAME).build()
                } catch (ignored: Exception) {
                    Log.v("AppDatabase", ignored.toString())
                }
            }
            return instance!!
        }
    }
}